programs/drift/src/state/user.rs

```diff
@@ -546,6 +549,118 @@ impl User {
     }
 }
 
+pub const RFQ_PDA_SEED: &str = "RFQ";
+#[zero_copy(unsafe)]
+#[derive(Default, Eq, PartialEq, Debug)]
+#[repr(C)]
+pub struct RFQOrderId {
+    pub uuid: [u8; 8],
+    pub max_ts: i64,
+}
+
+impl RFQOrderId {
+    pub fn new(uuid: [u8; 8], max_ts: i64) -> Self {
+        Self { uuid, max_ts }
+    }
+}
+
+impl Size for RFQUserAccount {
+    const SIZE: usize = 776;
+}
+
+#[account(zero_copy(unsafe))]
+#[derive(Default, Eq, PartialEq, Debug)]
+#[repr(C)]
+pub struct RFQUserAccount {
+    pub rfq_order_data: [RFQOrderId; 32],
+}
+
+impl RFQUserAccount {
+    pub fn check_exists_and_prune_stale_rfq_order_ids(
+        &mut self,
+        rfq_order_id: RFQOrderId,
+        now: i64,
+    ) -> DriftResult<bool> {
+        let mut uuid_exists: bool = false;
+        for i in 0..self.rfq_order_data.len() {
+            let existing_rfq_order_id = &mut self.rfq_order_data[i];
+            if existing_rfq_order_id.uuid == rfq_order_id.uuid && existing_rfq_order_id.max_ts > now
+            {
+                uuid_exists = true;
+            } else {
+                if existing_rfq_order_id.max_ts < now {
+                    existing_rfq_order_id.uuid = [0; 8];
+                    existing_rfq_order_id.max_ts = 0;
+                }
+            }
+        }
+        Ok(uuid_exists)
+    }
+
+    pub fn add_rfq_order_id(&mut self, rfq_order_id: RFQOrderId) -> DriftResult {
+        for i in 0..self.rfq_order_data.len() {
+            if self.rfq_order_data[i].max_ts == 0 {
+                self.rfq_order_data[i] = rfq_order_id;
+                return Ok(());
+            }
+        }
+
+        Err(ErrorCode::RFQUserAccountFull.into())
+    }
+}
+
+pub fn derive_rfq_user_pda(user_account_pubkey: &Pubkey) -> DriftResult<Pubkey> {
+    let (rfq_pubkey, _) = Pubkey::find_program_address(
+        &[RFQ_PDA_SEED.as_bytes(), user_account_pubkey.as_ref()],
+        &ID,
+    );
+    Ok(rfq_pubkey)
+}
+
+pub fn load_rfq_user_account_map<'a: 'b, 'b>(
+    account_info_iter: &mut Peekable<Iter<'a, AccountInfo<'b>>>,
+    user_pubkeys: Vec<Pubkey>,
+) -> DriftResult<BTreeMap<Pubkey, AccountLoader<'a, RFQUserAccount>>> {
+    let mut rfq_user_account_map = BTreeMap::<Pubkey, AccountLoader<'a, RFQUserAccount>>::new();
+
+    let available_maker_pdas: BTreeMap<Pubkey, Pubkey> = user_pubkeys
+        .iter()
+        .map(|pubkey| {
+            let pda = derive_rfq_user_pda(pubkey).unwrap();
+            (pda, *pubkey)
+        })
+        .collect();
+
+    for account_info in account_info_iter {
```

Comment:
i think if you add the auth/user key to the account, you can iterate through these and just check the discriminator like the other maps